tests/lib/rules/no-self-compare.js

```diff
@@ -21,10 +21,9 @@ const ruleTester = new RuleTester();
 ruleTester.run("no-self-compare", rule, {
     valid: [
         "if (x === y) { }",
-        "if (f() === f()) { }",
-        "if (a[1] === a[1]) { }",
```

Comment:
These tests were added in https://github.com/eslint/eslint/commit/fdcabe5da38bf44f7b6b4a1fc73b5d3007e6e911. It looks like they were intended to address the bug described in https://github.com/eslint/eslint/pull/160#r5439185, which is a separate case from what this issue addresses.